selfdrive/car/subaru/carstate.py

```diff
@@ -53,9 +53,13 @@ def update(self, cp, cp_cam, cp_body):
     steer_threshold = 75 if self.CP.carFingerprint in PREGLOBAL_CARS else 80
     ret.steeringPressed = abs(ret.steeringTorque) > steer_threshold
 
-    cp_cruise = cp_body if self.car_fingerprint in GLOBAL_GEN2 else cp
-    ret.cruiseState.enabled = cp_cruise.vl["CruiseControl"]["Cruise_Activated"] != 0
-    ret.cruiseState.available = cp_cruise.vl["CruiseControl"]["Cruise_On"] != 0
+    if self.car_fingerprint == CAR.FORESTER_2022:
+      ret.cruiseState.enabled = cp_cam.vl["ES_Status"]['Cruise_Activated'] != 0
+      ret.cruiseState.available = cp_cam.vl["ES_DashStatus"]['Cruise_On'] != 0
+    else:
+      cp_cruise = cp_body if self.car_fingerprint in GLOBAL_GEN2 else cp
+      ret.cruiseState.enabled = cp_cruise.vl["CruiseControl"]["Cruise_Activated"] != 0
+      ret.cruiseState.available = cp_cruise.vl["CruiseControl"]["Cruise_On"] != 0
```

Comment:
Is there no signal that we can use on all cars? If not, what other cars are going to be like this?

Reply:
Currently only Forester 2022 is confirmed, Crosstrek Hybrid is possible candidate but I have not been able to get a route yet